File: Section.py
from math import sqrt

class Section():
    def __init__(self, CRN, course, professor, building, timeSlots):
        #Course.__init__(self, subject, numID)
        self.CRN = CRN #int, the CRN of the section
        self.course = course #Course object
        self.professor = professor #Professor object
        self.GPA = professor.avgCourseGPA.get(course.ID,sqrt(professor.avgGPA*course.avgGPA)) #float, expected GPA of this section
        self.building = building #string, building the section is in
        self.timeSlots = timeSlots #list of Agenda objects




    def __eq__(self,obj):
        return isinstance(obj,Section) and self.CRN == obj.CRN

File: CODE/util/Section.js


class Section {
    constructor (CRN, course, professor, building, timeSlots) {
        this.CRN = CRN // int, the CRN of the section
        this.course = course // Course object
        this.professor = professor // Professor object
        this.GPA = _.get(professor.avgCourseGPA,[course.ID],Math.sqrt(professor.avgGPA*course.avgGPA))
        //(professor.avgCourseGPA[course.ID] * professor.avgGPA) ** (1/2) // float, expected GPA of this section
        // this.GPA = professor.avgCourseGPA.get(course.ID,(professor.avgGPA*course.avgGPA) ** (1/2)) // float, expected GPA of this section
        this.building = building // string, building the section is in
        this.timeSlots = timeSlots // list of Agenda objects
        course.sections.push(this)
    }

    equals (o) {
        return (o instanceof Section) && (this.CRN == o.CRN);
    }

    toString (o) {
        return `(${this.course}, ${this.CRN})`
    }
}
